server/plugins/org.eclipse.glsp.ide.editor/src/org/eclipse/glsp/ide/editor/ui/ChromiumKeyBindingFunction.java

```diff
@@ -97,11 +98,14 @@ protected Optional<Binding> findMatchingBinding(final IBindingService bindingSer
 
    protected void executeBinding(final Binding binding) {
       if (binding != null && binding.getParameterizedCommand().getCommand().isEnabled()) {
-         try {
-            binding.getParameterizedCommand().executeWithChecks(null, null);
-         } catch (ExecutionException | NotDefinedException | NotEnabledException | NotHandledException exception) {
-            LOGGER.error(exception);
-         }
+         Runnable runnable = () -> {
+            try {
+               binding.getParameterizedCommand().executeWithChecks(null, null);
+            } catch (ExecutionException | NotDefinedException | NotEnabledException | NotHandledException exception) {
+               LOGGER.error(exception);
+            }
+         };
+         Display.getDefault().asyncExec(runnable);
```

Comment:
Change looks good to me but could you please use the following utility method: `UIUtil.asyncExec`